.github/workflows/ompi_mpi4py.yaml

```diff
@@ -118,3 +167,135 @@ jobs:
       run:  python demo/test-run/test_run.py -v
       if:   ${{ true }}
       timeout-minutes: 10
+
+  #==============================================
+
+  run_spawn:
+    # This whole set of tests runs explicitly with setting "enable the
+    # spawn tests".  As of March 2024, we know that Open MPI is
+    # failing these tests.
+    needs: [build]
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+    - name: Use Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3
+        architecture: x64
+    - name: Get artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: /
+        name: build-artifacts
+    - name: Restore executable permissions
+      run: chmod a+x /opt/openmpi/bin/*
+    - name: Add Open MPI to PATH
+      run: echo /opt/openmpi/bin >> $GITHUB_PATH
+    - name: Install the mpi4py wheel
+      run: python -m pip install mpi4py --no-index --find-links=.
+
+    #----------------------------------------------
+
+    - name: Test mpi4py (singleton)
+      run:  python test/main.py -v
+      if:   ${{ true }}
+      env:
+        MPI4PY_TEST_SPAWN: "1"
```

Comment:
@jsquyres Minor nit: You can add an `env` section at the job level. In there, you set `MPI4PY_TEST_SPAWN: "1"`. That way you do not have to add the environment variable on each individual step.

Reply:
Look at the suggestion I added above. Of course, you should do the same with the mpi4py-dynamic job.